packages/ccd-js-gen/jest.config.ts

```diff
@@ -0,0 +1,27 @@
+import type { Config } from 'jest';
+import type {} from 'ts-jest';
+
+const config: Config = {
+    preset: 'ts-jest',
+    testEnvironment: 'node',
+    moduleFileExtensions: ['js', 'ts', 'json'],
+    moduleDirectories: ['node_modules'],
+    moduleNameMapper: {
+        '^(\\.\\.?\\/.+)\\.js$': '$1', // Remap esmodule file extensions
+    },
+    transformIgnorePatterns: [
+        'node_modules/(?!@noble/ed25519)', // @noble/ed25519 is an ES module only
+    ],
+    transform: {
+        '^.+\\.[jt]sx?$': [
+            'ts-jest',
+            {
+                useESM: true,
+                tsconfig: 'tsconfig.json',
+            },
+        ],
+    },
+    extensionsToTreatAsEsm: ['.ts'],
```

Comment:
```suggestion
    preset: 'ts-jest/presets/js-with-ts-esm',
    moduleNameMapper: {
        '^(\\.\\.?\\/.+)\\.js$': '$1', // Remap esmodule file extensions
    }
```

This would be a similar, more minimal config (I tinkered with this for the SDK tests on another branch). I assumed here, that you don't depend on `@noble/ed25519`, but if you do then obviously that should still be transformed.